pkg/spi-shared/config/config.go

```diff
@@ -57,7 +73,28 @@ type CommonCliArgs struct {
 // consumption.
 type persistedConfiguration struct {
 	// ServiceProviders is the list of configuration options for the individual service providers
-	ServiceProviders []ServiceProviderConfiguration `yaml:"serviceProviders"`
+	ServiceProviders []persistedServiceProviderConfiguration `yaml:"serviceProviders"`
+}
+
+// ServiceProviderConfiguration contains configuration for a single service provider configured with the SPI. This
+// mainly contains config.yaml of the OAuth application within the service provider.
+type persistedServiceProviderConfiguration struct {
+	// ClientId is the client ID of the OAuth application that the SPI uses to access the service provider.
+	ClientId string `yaml:"clientId"`
+
+	// ClientSecret is the client secret of the OAuth application that the SPI uses to access the service provider.
+	ClientSecret string `yaml:"clientSecret"`
+
+	// ServiceProviderType is the type of the service provider. This must be one of the supported values: GitHub, Quay
+	ServiceProviderName ServiceProviderName `yaml:"type"`
+
+	// ServiceProviderBaseUrl is the base URL of the service provider. This can be omitted for certain service provider
+	// types, like GitHub that only can have 1 well-known base URL.
+	ServiceProviderBaseUrl string `yaml:"baseUrl,omitempty"`
+
+	// Extra is the extra configuration required for some service providers to be able to uniquely identify them. E.g.
+	// for Quay, we require to know the organization for which the OAuth application is defined for.
```

Comment:
The mention about Quay is incorrect, let's just delete it from the comment when we're at it? 😃

Reply:
fixed